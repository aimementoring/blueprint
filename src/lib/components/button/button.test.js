import React from 'react';
import renderer from 'react-test-renderer';
import Button from './button';

describe('Button', () => {
  it('is backwards compatible (using text prop instead of children)', () => {
    const tree = renderer
      .create(<Button type="button" onClickFunction={() => {}} text="Back" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders button properly', () => {
    const tree = renderer
      .create(
        <Button type="button" onClickFunction={() => {}}>
          Back
        </Button>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders link properly', () => {
    const tree = renderer
      .create(
        <Button type="link" url="www.aimementoring.com">
          Link button
        </Button>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders button properly with underneath label', () => {
    const tree = renderer
      .create(
        <Button type="link" underneathLabel="My label">
          Button
        </Button>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
