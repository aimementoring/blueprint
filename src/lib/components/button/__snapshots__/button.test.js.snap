// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button is backwards compatible (using text prop instead of children) 1`] = `
<div
  className="theme-base"
>
  <div
    className=""
  >
    <div
      className="buttonGroup"
    >
      <button
        className=" button"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Back
      </button>
    </div>
  </div>
</div>
`;

exports[`Button renders button properly 1`] = `
<div
  className="theme-base"
>
  <div
    className=""
  >
    <div
      className="buttonGroup"
    >
      <button
        className=" button"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Back
      </button>
    </div>
  </div>
</div>
`;

exports[`Button renders button properly with underneath label 1`] = `
<div
  className="theme-base"
>
  <div
    className=""
  >
    <div
      className="buttonGroup"
    >
      <button
        className=" button"
        disabled={false}
        onClick={[Function]}
        type="link"
      >
        Button
      </button>
      <label
        className="underneathLabel"
      >
        My label
      </label>
    </div>
  </div>
</div>
`;

exports[`Button renders link properly 1`] = `
<div
  className="theme-base"
>
  <div
    className=""
  >
    <a
      className=" linkButton"
      href="http://www.aimementoring.com"
      target="_blank"
    >
      Link button
    </a>
  </div>
</div>
`;
